### PR TITLE
#27854 non existing folder skeletondirectory

### PR DESCRIPTION
**Pull request by Gandolf90**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: lib/private/legacy/util.php [hunk 1]
```diff
@@ -388,11 +388,16 @@ public static function getUserQuota($userId) {
...
 	 * @param String $userId
 	 * @param \OCP\Files\Folder $userDirectory
+	 * @throws \OC\HintException
 	 */
 	public static function copySkeleton($userId, \OCP\Files\Folder $userDirectory) {
 
 		$skeletonDirectory = \OCP\Config::getSystemValue('skeletondirectory', \OC::$SERVERROOT . '/core/skeleton');
 
+		if(!is_dir($skeletonDirectory))  {
+			throw new OC\HintException('Can not Access to Skeleton Folder, check for right Path in config.php');
```
Comment (DeepDiver1975) [inline lib/private/legacy/util.php]: - [ ] please add the value of $skeletonDirectory
- [ ] test should read as
```php
throw new OC\HintException("The skeleton folder '$skeletonFolder' is not accessible.');

File: tests/lib/UtilTest.php [hunk 2]
```diff
@@ -413,6 +414,19 @@ public function testCheckDataDirectoryValidity() {
 		$this->assertNotEmpty($errors);
 	}
 
+	public function testCopySkeletonDirectory() {
+		$this->expectException(HintException::class);
```
Comment (DeepDiver1975) [inline tests/lib/UtilTest.php]: please use phpunit annotation - see the phpunit docs

**Review by DeepDiver1975 — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution!
Very welcome!
Please make sure to squash our commits. THX

**Comment by DeepDiver1975** [comment]
Please also sign the CLA - see the comment above

**Comment by IljaN** [comment]
Please reference pull requests to issues like this #27854

File: tests/lib/UtilTest.php [hunk 3]
```diff
@@ -413,6 +414,27 @@ public function testCheckDataDirectoryValidity() {
...
 
+	/**
+	 * @expectedException HintException
+	 *
+	 * @throws HintException
+	 */
+	public function testCopySkeletonDirectory() {
+		$notExistingDirectory = '/not/existing/Directory';
+
+		$this->expectException(HintException::class);
```
Comment (IljaN) [inline tests/lib/UtilTest.php]: You can remove this line if you use the annotation see:
https://phpunit.de/manual/current/en/appendixes.annotations.html#appendixes.annotations.expectedExceptionMessage

File: lib/private/legacy/util.php [hunk 4]
```diff
@@ -388,11 +388,16 @@ public static function getUserQuota($userId) {
 	 *
 	 * @param String $userId
 	 * @param \OCP\Files\Folder $userDirectory
+	 * @throws \OC\HintException
 	 */
 	public static function copySkeleton($userId, \OCP\Files\Folder $userDirectory) {
 
 		$skeletonDirectory = \OCP\Config::getSystemValue('skeletondirectory', \OC::$SERVERROOT . '/core/skeleton');
 
+		if(!is_dir($skeletonDirectory))  {
```
Comment (IljaN) [inline lib/private/legacy/util.php]: Missing space between if and opening parenthesis => if ( )

**Comment by IljaN** [comment]
@Gandolf90 Some tests are failing, please fix. See Jenkins for details.

**Comment by Gandolf90** [comment]
@IljaN added changes 
new added tests passed - should i repair some other tests ?

**Comment by Gandolf90** [comment]
@IljaN how can I test the failed Selenium/Travis tests?

**Comment by IljaN** [comment]
@Gandolf90  "make test"

**Comment by Gandolf90** [comment]
@IljaN  testet local with "make test" and jenkins tests passed
don't now why travis make with php 5.6 settings and selenium error to connect with curl as user

**Comment by IljaN** [comment]
@Gandolf90 Don`t worry our CI is like a diva sometimes :)

**Comment by phil-davis** [comment]
@Gandolf90 Selenium/Travis fails because this is a PR from a forked copy of the repo. So Travis does not provide the job with the authentication keys to the ownCloud SauceLabs account. There are 2 options:
1) you can ask for access to the ownCloud core repo (which I got as a new community member a few years ago without any hassle) - then clone that and make your own branch in there.
2) create yourself a SauceLabs open-source free account, then put the details in your own Travis linked to GitHub. Then make a PR of the code in your fork to the master of your fork - Travis and SauceLabs will happily talk using all your own authentication data.

**Comment by Gandolf90** [comment]
@IljaN  or @DeepDiver1975  can you create a temporary access to make the pull request ?

**Comment by Gandolf90** [comment]
@phil-davis  thanks for Hint

**Comment by IljaN** [comment]
@Gandolf90  Why close?

**Comment by Gandolf90** [comment]
@iljan Because i wanted to make a Branch from Core Repo and make there the pr as @phil-davis wrote

**Comment by phil-davis** [comment]
@Gandolf90 when you make the PR in core, please mention it here so it is easy to find.

**Comment by Gandolf90** [comment]
@phil-davis ok

**Comment by Gandolf90** [comment]
@IljaN  @DeepDiver1975  @phil-davis  here it goes : https://github.com/owncloud/core/pull/29367

**Pull request closed without merging**